pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -632,4 +638,29 @@ export class ExtensionImpl implements Extension {
   async readFile(sourcePath: string): Promise<string> {
     return await this.client.readFile(this.image, sourcePath, { namespace: this.extensionNamespace });
   }
+
+  async shutdown() {
```

Comment:
Should this function have some kind of timeout mechanism so it cannot block application shutdown indefinitely (I assume the app will wait for all hooks to terminate before shutting down).

Reply:
We don't actually wait for the hooks to terminate at all. It's only async to make sure we can get metadata (before we spawn the extension-specified commands).